Task #1

In [21]:
import pandas as pd

airports = pd.read_csv('https://raw.githubusercontent.com/hadley/nycflights13/master/data-raw/airports.csv')
weather = pd.read_csv('https://raw.githubusercontent.com/hadley/nycflights13/master/data-raw/weather.csv')

airports['lat'].isnull().sum()
northernmost = airports.sort_values(by='lat', ascending=False).head(5)
northernmost[['faa', 'name', 'lat', 'lon']]

faa                                        name        lat         lon
417  EEN                     Dillant Hopkins Airport  72.270833   42.898333
230  BRW                  Wiley Post Will Rogers Mem  71.285446 -156.766003
110  AIN                          Wainwright Airport  70.638056 -159.994722
708  K03                               Wainwright As  70.613378 -159.860350
152  ATK  Atqasuk Edward Burnell Sr Memorial Airport  70.467300 -157.436000

Based on the dataset provided in airports.csv, the northernmost airport is Dillant-Hopkins Airport (EEN), which appears with a latitude of 72.3° N. This result was determined by sorting the dataset in descending order by latitude and selecting the top entry. No latitude values were missing or required cleaning, and the analysis was performed without filtering by state or region. The result reflects the internal structure of the dataset rather than verified geographic positioning.

Task #2

In [22]:
airports['lon'].isnull().sum()
easternmost = airports.sort_values(by='lon', ascending=False).head(5)
easternmost[['faa', 'name', 'lat', 'lon']]

faa                           name        lat         lon
1290  SYA                   Eareckson As  52.712275  174.113620
942   MYF               Montgomery Field  32.475900  117.759000
396   DVT  Deer Valley Municipal Airport  33.411700  112.457000
417   EEN        Dillant Hopkins Airport  72.270833   42.898333
444   EPM     Eastport Municipal Airport  44.910111  -67.012694

According to the dataset, the easternmost airport is Eareckson AS (SYA), which has a longitude of 174.11 . This was determined by sorting the airports.csv file in descending order by longitude, since less negative values indicate locations farther east. No longitude values were missing, and the analysis included all airports listed in the dataset, including those in U.S. territories. The result reflects the dataset’s internal coordinates and does not exclude overseas airports, as they are valid entries under the assignment’s scope.

Task #3

In [19]:
weather['date'] = pd.to_datetime(weather[['year', 'month', 'day']])

feb12 = weather[weather['date'] == '2013-02-12']

nyc_airports = ['JFK', 'LGA', 'EWR']
feb12_nyc = feb12[feb12['origin'].isin(nyc_airports)]

feb12_nyc_clean = feb12_nyc[feb12_nyc['wind_speed'] < 100]

windiest = feb12_nyc_clean.groupby('origin')['wind_speed'].max().sort_values(ascending=False)
print("Max wind speeds on Feb 12, 2013:")
print(windiest)

Max wind speeds on Feb 12, 2013:
origin
LGA    23.01560
EWR    21.86482
JFK    20.71404
Name: wind_speed, dtype: float64


On February 12th, 2013, the New York area airport with the highest recorded wind speed was LaGuardia Airport (LGA), with a maximum wind speed of 23.02 mph. This was determined by filtering the weather.csv file for that specific date and selecting only the three NYC airports: JFK, LGA, and EWR. The data was grouped by airport and the maximum wind speed was calculated for each. During the initial analysis, EWR showed a wind speed over 1000 mph, which was clearly a data error. To correct for this, any wind speed values above 100 mph were excluded from the final calculation. No interpolation or imputation was applied, and the dataset contained sufficient valid entries to support the result. This highlights the importance of checking for outliers when working with real-world data.
